image_pyramid.py

```diff
@@ -84,23 +110,29 @@ def get_submatrix(self, top_left_x, top_left_y):
         min_y >>= scale
         max_x >>= scale
         max_y >>= scale
-        # Avoid roundoff errors: make the truncated edges 1 pixel wider before
-        # expanding, and if it's a few pixels larger than expected, no one
-        # will notice.
+        # To make roundoff errors harder to notice, make the truncated edges 1
+        # pixel wider before expanding. Better to be too big than too small.
         max_x += 1
         max_y += 1
 
         submatrix = current_data[min_y:max_y, min_x:max_x]
+        image = numpy.zeros([*submatrix.shape, 3], numpy.uint8)
+        image[:, :, 2] = submatrix * 255
+
+        if self._hue_pyramid is not None:
+            image[:, :, 0] = self._hue_pyramid[0][min_y:max_y, min_x:max_x]
+            image[:, :, 1] = 255
```

Comment:
This all is duplicated above and again in visual_diff.py. Make it a helper function.